src/v/http/client.cc

```diff
@@ -589,4 +590,25 @@ ss::input_stream<char> client::response_stream::as_input_stream() {
     return ss::input_stream<char>(std::move(ds));
 }
 
+client::request_header redacted_header(client::request_header original) {
+    using field_type = std::variant<boost::beast::http::field, std::string>;
+
+    static const std::unordered_set<field_type> redacted_fields{
+      boost::beast::http::field::authorization, "x-amz-content-sha256"};
+
+    auto h{std::move(original)};
+
+    for (const auto& field : redacted_fields) {
+        std::visit(
+          [&h](const auto& f) {
+              if (h.find(f) != h.end()) {
+                  h.set(f, "*****");
```

Comment:
nit: make  "*****" const? Also we have `static constexpr std::string_view secret_placeholder = "[secret]";` for properties, maybe we could use something similar

Reply:
good point, thanks, I will use the same value.